src/palace/manager/integration/license/overdrive/importer.py

```diff
@@ -0,0 +1,323 @@
+from palace.manager.util.datetime_helpers import datetime_utc
+from palace.manager.util.log import LoggerMixin
+
+
+@dataclass(frozen=True)
+class FeedImportResult:
+    current_page: BookInfoEndpoint
+    next_page: BookInfoEndpoint | None = None
+    processed_count: int = 0
+
+
+class OverdriveImporter(LoggerMixin):
+    DEFAULT_START_TIME = datetime_utc(1970, 1, 1)
+    DEFAULT_PAGE_SIZE = 100
+
+    def __init__(
+        self,
+        db: Session,
+        collection: Collection,
+        registry: LicenseProvidersRegistry,
+        import_all: bool = False,
+        identifier_set: IdentifierSet | None = None,
+        parent_identifier_set: IdentifierSet | None = None,
+        api: OverdriveAPI | None = None,
+    ) -> None:
+        """Constructor for the OverdriveImporter class.
+
+        Args:
+            db: The database session.
+            collection: The collection to import.
+            registry: The license providers registry.
+            import_all: Whether to import all books from the collection.
+            identifier_set: The identifier set to use for the import.
+            parent_identifier_set: The parent identifier set to use for the import.
+            api: The OverdriveAPI instance to use for the import.
+        """
+        self._db = db
+        self._collection = collection
+        self._import_all = import_all
+        self._identifier_set = identifier_set
+
+        self._parent_identifiers: Set[str] | None = None
+        if parent_identifier_set is not None:
+            # create an in-memory set from the redis set to optimize existence checks for individual identifiers.
+            # I don't believe we need to worry about memory here: few redis identifier sets will likely exceed 200K
+            # items which should be easily manageable given an identifier is 36 characters (36*200K = 7.2 MB). Most OD
+            # collections are much  smaller in the 20-70K range.
+            self._parent_identifiers = {
+                x.identifier for x in parent_identifier_set.get()
+            }
+
+        if not registry.equivalent(collection.protocol, OverdriveAPI):
+            raise PalaceValueError(
+                f"Collection {collection.name} [id={collection.id} protocol={collection.protocol}] "
+                f"is not an OverDrive collection."
+            )
+
+        self._api = (
+            OverdriveAPI(_db=self._db, collection=self._collection)
+            if api is None
+            else api
+        )
+
+        self._extractor = OverdriveRepresentationExtractor(self._api)
+
+    def get_timestamp(self) -> Timestamp:
+        timestamp, _ = get_one_or_create(
+            self._db,
+            Timestamp,
+            service="OverDrive Import",
+            service_type=Timestamp.TASK_TYPE,
+            collection=self._collection,
+        )
+        return timestamp
+
+    def _process_book(
+        self,
+        book: dict[str, Any],
```

Comment:
Minor: I know this comes from the existing importers, but it would be much easier to reason about whats in a book if we were using a dataclass or a pydantic model to represent a book. This has been a source of bugs from the original overdrive importer. Its outside the scope here, but it would have been nice to sort this out when moving to the new importers.